From 21b8d48cb683118ff9fbcbd1d077a52e602a4bb7 Mon Sep 17 00:00:00 2001
Message-Id: <21b8d48cb683118ff9fbcbd1d077a52e602a4bb7.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Rafael Barbalho <rafael.barbalho@intel.com>
Date: Thu, 17 Jul 2014 18:38:49 +0100
Subject: [PATCH 189/312] FOR_UPSTREAM [VPG]: drm/i915: Fix PPGTT PDP
 allocations

On systems that the GTT is only 2GB, i.e. cherryview, we are not
allocating enough PDP tables to fill out the full 4GB PPGTT address space.

Change-Id: Ibe4277640a055338c0fe015530323334cb82429d
Signed-off-by: Rafael Barbalho <rafael.barbalho@intel.com>
---
 drivers/gpu/drm/i915/i915_gem_gtt.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_gtt.c b/drivers/gpu/drm/i915/i915_gem_gtt.c
index 93ecc66..9c81f0b 100644
--- a/drivers/gpu/drm/i915/i915_gem_gtt.c
+++ b/drivers/gpu/drm/i915/i915_gem_gtt.c
@@ -605,7 +605,7 @@ static int gen8_ppgtt_init(struct i915_hw_ppgtt *ppgtt, uint64_t size)
 		DRM_INFO("Pages will be wasted unless GTT size (%llu) is divisible by 1GB\n", size);
 
 	/* 1. Do all our allocations for page directories and page tables. */
-	ret = gen8_ppgtt_alloc(ppgtt, max_pdp);
+	ret = gen8_ppgtt_alloc(ppgtt, GEN8_LEGACY_PDPS);
 	if (ret)
 		return ret;
 
-- 
1.7.9.5

